pipeline/transformer/transformer_calm/src/test/scala/weco/pipeline/transformer/calm/transformers/CalmMergeCandidatesTest.scala

```diff
@@ -0,0 +1,90 @@
+package weco.pipeline.transformer.calm.transformers
+
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import weco.catalogue.internal_model.identifiers.{
+  IdState,
+  IdentifierType,
+  SourceIdentifier
+}
+import weco.catalogue.internal_model.work.MergeCandidate
+import weco.catalogue.source_model.generators.CalmRecordGenerators
+
+class CalmMergeCandidatesTest
+    extends AnyFunSpec
+    with Matchers
+    with CalmRecordGenerators {
+
+  it("creates a Sierra mergeCandidate from the BNumber field") {
+    val bnumber = "b1234567"
```

Comment:
This is an interesting example, because if this was the value we wouldn't be able to merge the b-number.

Sierra b-numbers are presented in one of two flavours:

* Sierra identifier: seven digits (e.g. `1234567`), or
* Sierra system number: eight digits with a prefix (e.g. `b12345672`)

but never seven digits and a prefix.

A quick skim suggests that the Calm catalogue mostly uses the eight-digit-with-prefix variant. This is another reason we might want to be stricter about what we use as merge candidates.

(I realise this is an existing problem, but since you're here I'm going to mention it.)

Reply:
Going to address this and the other validation comment after https://github.com/wellcomecollection/catalogue-pipeline/pull/1746